src/mds/Server.cc

```diff
@@ -11994,6 +12045,64 @@ bool Server::build_snap_diff(
 }
 
 #ifdef WITH_CEPHFS_NOTIFICATION
+void Server::handle_client_add_kafka_topic(const MDRequestRef &mdr) {
+  const cref_t<MClientRequest> &req = mdr->client_request;
+  KafkaTopicPayload payload;
+  if (req->get_data().length()) {
+    auto iter = req->get_data().cbegin();
+    decode(payload, iter);
+  }
```

Comment:
Return error if no or undecodable payload found?

Reply:
Later we might want to introduce spefic user rights for being able to control notification endpoints and check them here and there. Perhaps it's worth to have "FIXME" comments in the new methods about that.